transports/quic/src/transport.rs

```diff
@@ -188,7 +204,9 @@ impl<P: Provider> Transport for GenTransport<P> {
         let (socket_addr, version, _peer_id) = self.remote_multiaddr_to_socketaddr(addr, false)?;
         let endpoint_config = self.quinn_config.endpoint_config.clone();
         let server_config = self.quinn_config.server_config.clone();
-        let socket = UdpSocket::bind(socket_addr).map_err(Self::Error::from)?;
+        let socket = self.create_socket(socket_addr).map_err(Self::Error::from)?;
+
+        let socket: UdpSocket = socket.into();
```

Comment:
You could directly return an std socket from the fn, then the use of `socket2` would be encapsulated.